File: rust_study/cpp/Leetcode/128.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace Leetcode\128.cpp
 *@data      2023/12/25 13:39:22
 */
#include<bits/stdc++.h>
#include<unordered_set>
#include<unordered_map>
#define ll long long

using namespace std;

int longestConsecutive(vector<int> &nums) {
    unordered_set<int> num_set;
    for(auto &x : nums)num_set.insert(x);
    int ans = 0;
    for(auto &x : num_set) {
        if(!num_set.count(x - 1)) {
            int now = x, len = 1;
            while(num_set.count(++now))++len;
            ans = max(ans, len);
        }
    }
    return ans;
}
void solve() {
    int n, m, i, j;
    vector<int> qwq = { 1, 2, 3, 4, 5, 6, 7 };
    cout << longestConsecutive(qwq);
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_leetcode/src/bin/128.rs
use std::cmp::Ordering;
use std::collections::BinaryHeap;

//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace leetcode/128.rs
//@data      2023/12/24 19:10:26

fn main() {
    let t = 1;
    //let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| {
        println!("{}", solve(vec![0, 1, 1, 2]));
    });
}

pub fn solve(nums: Vec<i32>) -> i32 {
    if nums.is_empty() {
        return 0;
    }
    let mut nums = BinaryHeap::from(nums);
    let (mut ans, mut now) = (1, 1);
    let mut qaq = nums.pop().unwrap();
    while !nums.is_empty() {
        let qwq = nums.pop().unwrap();
        match qwq.cmp(&(qaq - 1)) {
            Ordering::Equal => now += 1,
            Ordering::Less => now = 1,
            Ordering::Greater => {}
        }
        qaq = qwq;
        ans = ans.max(now);
    }
    ans
}
